daemon/cmd/cells.go

```diff
@@ -110,6 +110,7 @@ var (
 		// Provides optional configuration callback to bypass
 		// host firewall when accessing Kubernetes objects.
 		hostfirewallbypass.Cell,
+		cell.Invoke(hostfirewallbypass.RestartClientset),
```

Comment:
This should probably live inside the Cell for host firewall rather than in the daemon.

Reply:
I'll add that I'm not sure this is the right decision to enable by default. I gather that it's a simpler solution to the underlying problem but I worry about the impact on other users.

Reply:
What's not clear to me for instance is (a) how much additional k8s state might be retransmitted to agents due to this, and (b) how the timing of this competes with the agent's startup and sync of all state from the control plane. Background for (b) being that Cilium expects to start up, sync all state with k8s, then start applying changes based on the newly discovered desired state.

Reply:
> I'll add that I'm not sure this is the right decision to enable by default.
Me neither. I just enabled it in this draft PR so that CI tests use it.

Reply:
Maybe naive question, why would `cilium-dbg build-config` include this Cell?

Reply:
The bypass cell? Because `build-config` init container executes while the daemon (and therefore the DNS proxy) is not running, but the BPF programs or iptables rules directing host DNS requests to the proxy are up. Without the bypass the `build-config` init container fails and the whole node becomes unusable.